include/boost/json/basic_parser.hpp

```diff
@@ -734,30 +734,32 @@ parse_string(
         start = cs.data();
         state st;
         st_.pop(st);
-        switch(st)
+        if (st != state::str1)
```

Comment:
This is now irregular (different from all the other switch-restore stanzas in the file).

Reply:
The alternative is the keep `state::str1` and manually call trim at each case label